src/System.Configuration.ConfigurationManager/tests/System/Configuration/TempConfigurationHost.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Configuration;
+using System.Configuration.Internal;
+using System.Reflection;
+
+public class TempConfigurationHost : DelegatingConfigHost {
+    static string assemblyName = PlatformDetection.IsFullFramework ? "System.Configuration, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" : "System.Configuration.ConfigurationManager";
+    static IInternalConfigConfigurationFactory s_configurationFactory;
+    
+    private ConfigurationFileMap m_fileMap;
+
+    public TempConfigurationHost() {
+        Type type = Type.GetType(InternalHostTypeName, true);
+        Host = (IInternalConfigHost) Activator.CreateInstance(type, true);
+    }
+
+    public override void Init(IInternalConfigRoot configRoot, params object[] hostInitParams) {
+        Host.Init(configRoot, hostInitParams);
+    }
+
+    public override void InitForConfiguration(ref string locationSubPath, out string configPath, out string locationConfigPath,
+                        IInternalConfigRoot configRoot, params object[] hostInitConfigurationParams) {
+
+        Host.Init(configRoot, hostInitConfigurationParams);
+
+        m_fileMap = hostInitConfigurationParams[1] as ConfigurationFileMap;
+        
+        locationSubPath = ConfigurationFactory.NormalizeLocationSubPath(locationSubPath, null);
+        configPath = "MACHINE/EXE";
+        locationConfigPath = locationSubPath;
+
+        
+    }
+
+    public override bool IsTrustedConfigPath(string configPath) {
+        return true;
+    }
+
+    public override bool IsLocationApplicable(string configPath) {
+        return true;
+    }
+
+    public override bool IsFullTrustSectionWithoutAptcaAllowed(IInternalConfigRecord configRecord) {
+        return true;
+    }
+
+    public override bool PrefetchAll(string configPath, string streamName)
+    {
+        return false;
+    }
+
+    public override bool PrefetchSection(string sectionGroupName, string sectionName)
+    {
+        return false;
+    }
+
+    public override string GetStreamName(string configPath) {
+        return m_fileMap.MachineConfigFilename;
+    }
+
+    static string InternalConfigConfigurationFactoryTypeName {
+        get {
+            return string.Format("System.Configuration.Internal.InternalConfigConfigurationFactory, {0}", assemblyName);
+        }
+    }
+
+    static string InternalHostTypeName {
+        get {
+            return string.Format("System.Configuration.Internal.InternalConfigHost, {0}", assemblyName);
```

Comment:
Nit: earlier you just used concatenation for the same kind of thing:
https://github.com/dotnet/corefx/pull/30548/files#diff-fa83fd05ca83abe5ad010b52927507b6R74
It'd be nice to be consistent.